integrationtests/self/http_test.go

```diff
@@ -140,6 +140,24 @@ var _ = Describe("HTTP tests", func() {
 		Expect(resp.Header.Get("Content-Length")).To(Equal(strconv.Itoa(len("foobar"))))
 	})
 
+	It("detects stream errors when server panics when writing response", func() {
+		mux.HandleFunc("/writing_and_panicking", func(w http.ResponseWriter, r *http.Request) {
+			// no recover here as it will interfere with the handler
+			w.Write([]byte("foobar"))
+			w.(http.Flusher).Flush()
+			// simulate a large response body, some data were written
+			time.Sleep(5 * time.Second)
+			panic(http.ErrAbortHandler)
+		})
+
+		resp, err := client.Get(fmt.Sprintf("https://localhost:%d/writing_and_panicking", port))
+		Expect(err).ToNot(HaveOccurred())
+		body, err := io.ReadAll(resp.Body)
+		Expect(err).To(HaveOccurred())
+		// in reality the parts of the body is undetermined, but it's a test
+		Expect(body).To(Equal([]byte("foobar")))
```

Comment:
Why do you test for something that's undetermined?

Reply:
Should I change it so it just tests there are data that are just prefix of the data we just wrote?

Reply:
All that you can assert is that the error code is correct, right? The stream data might have been lost in transit.

Reply:
Some data might have been lost, but those that have been received are sure be the prefix of what we had written (`nil` is prefix of the data). Without the `time.Sleep` the client won't even receive the response.